laws/src/main/scala/cats/laws/AlternativeLaws.scala

```diff
@@ -0,0 +1,25 @@
+package cats.laws
+
+import cats.Alternative
+
+import cats.syntax.all._
+
+trait AlternativeLaws[F[_]] extends ApplicativeLaws[F] with MonoidKLaws[F] {
+  implicit override def F: Alternative[F]
+  implicit def algebra[A] = F.algebra[A]
+
+  def alternativeRightAbsorption[A, B](ff: F[A => B]): IsEq[F[B]] =
+    (F.empty[A] apply ff) <-> F.empty[B]
+
+  def alternativeLeftDistributivity[A, B](fa: F[A], fa2: F[A], f: A => B): IsEq[F[B]] =
+    ((fa |+| fa2) map f) <-> ((fa map f) |+| (fa2 map f))
+
+  def alternativeRightDistributivity[A, B](fa: F[A], ff: F[A => B], fg: F[A => B]): IsEq[F[B]] =
+    (fa apply (ff |+| fg)) <-> ((fa apply ff) |+| (fa apply fg))
```

Comment:
What is `right distributivity`? where came from?

it does not exists in Haskell https://hackage.haskell.org/package/base-4.8.0.0/docs/src/GHC-Base.html#Alternative

```
-- | A monoid on applicative functors.
--
-- If defined, 'some' and 'many' should be the least solutions
-- of the equations:
--
-- * @some v = (:) '<$>' v '<*>' many v@
--
-- * @many v = some v '<|>' 'pure' []@
```

and `XorT` does not satisfy this law.

- https://github.com/non/cats/pull/287
- https://github.com/non/cats/commit/c11a1bfe880b4e445e85cd2c34579

I think `right distributivity` is not necessary.